### ISSUE TITLE
The inference speed is slower with Tensort c++ than with pytorch

### ISSUE THREAD

**Issue by LinuxCup** [NVIDIA/TensorRT#3097]
## Environment

**TensorRT Version**:
8.5。1.7
**NVIDIA GPU**:
P4000
**NVIDIA Driver Version**:
525.105.17
**CUDA Version**:
11.8
**CUDNN Version**:
8.6

Operating System:
ubuntu 18.04
Python Version (if applicable):
3.8
Tensorflow Version (if applicable):

PyTorch Version (if applicable):
1.13.1
Baremetal or Container (if so, version):

## Relevant Files

**Model link**:

## Steps To Reproduce

**Commands or scripts**:

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**:

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`):

**Comment by LinuxCup** [comment]
reference example(sampleDynamicReshape) I used dynamic shape for model input, Model inference time is about 256ms. But inference with pytorch is much less time elapse, about 140ms. Why the time takes more with TensorRT c++ code? 
Looking forward to reply! Some debugging suggestions!

**Comment by zerollzeng** [comment]
> double t4 = getTime();
>     bool status = context_->executeV2(processorBindings.data());
>     if (!status)
>     {
>         return false;
>     }
>     cudaDeviceSynchronize();
>     double t5 = getTime();

run this for few more times, e.g. 100 iterations, because the first run will need to initialize some resource.
Don't use cudaDeviceSynchronize(), use stream synchronize.

**Comment by LinuxCup** [comment]
yes,I followed your suggestion and iterated many times, the average time cost as follows:
t1-t0:0.00903893 (unit: s)
t2-t1:0.0125654
t3-t2:3.98159e-05
t4-t3:6.4373e-06
t5-t4:0.286378
t6-t5:0.00112915
-------------------------------------------------------------
the result is the same as before. however,the pytorch version time cost about 140ms.

**Comment by LinuxCup** [comment]
execute the following bash script:
./trtexec --onnx=dsvt_blocks.onnx \
--saveEngine=dsvt_blocks.engine \
--memPoolSize=workspace:4096 --verbose --buildOnly --device=0 --fp16 \
--tacticSources=+CUDNN,+CUBLAS,-CUBLAS_LT,+EDGE_MASK_CONVOLUTIONS \
minShapes=src:1000x128,set_voxel_inds_tensor_shift_0:2x50x36,set_voxel_inds_tensor_shift_1:2x50x36,set_voxel_masks_tensor_shift_0:2x50x36,set_voxel_masks_tensor_shift_1:2x50x36,pos_embed_tensor:4x2x1000x128 \
optShapes=src:24629x128,set_voxel_inds_tensor_shift_0:2x1156x36,set_voxel_inds_tensor_shift_1:2x834x36,set_voxel_masks_tensor_shift_0:2x1156x36,set_voxel_masks_tensor_shift_1:2x834x36,pos_embed_tensor:4x2x24629x128 \
maxShapes=src:100000x128,set_voxel_inds_tensor_shift_0:2x5000x36,set_voxel_inds_tensor_shift_1:2x3200x36,set_voxel_masks_tensor_shift_0:2x5000x36,set_voxel_masks_tensor_shift_1:2x3200x36,pos_embed_tensor:4x2x100000x128

the results output file is as follows:
[result.txt](https://github.com/NVIDIA/TensorRT/files/11954937/result.txt)

**Comment by zerollzeng** [comment]
remove `--buildOnly` and see how TRT give a perf summary.

**Comment by LinuxCup** [comment]
> remove `--buildOnly` and see how TRT give a perf summary.

Thank you very much! The engine trt file is loaded in a way that works. performance sunmary is as follows
time:0.00162653
time:0.00199061
time:6.76522e-06
time:1.22136e-06
time:0.0427477
time:0.000196134
1. why does is work when remove `buildonly`?
2. by the way, the another way by parsing onnx file, How to code with c++ ?

**Comment by LinuxCup** [comment]
it seems to remain unchanged, the performance summary is as follow:
time:0.00957238
time:0.0118158
time:4.06106e-05
time:7.2404e-06
time:0.261589
time:0.00116311
is it related to the gpu hardware?

**Comment by LinuxCup** [comment]
print some sensitive information when trtexec script is executed, as:

> [07/07/2023-11:02:26] [W] [TRT] Myelin graph with multiple dynamic values may have poor performance if they differ. Dynamic values are: 
> [07/07/2023-11:02:26] [W] [TRT]  voxel_number
> [07/07/2023-11:02:26] [W] [TRT]  set_number_shift_1
> [07/07/2023-11:02:26] [W] [TRT]  set_number_shift_0
> [07/07/2023-11:04:53] [I] [TRT] Total Activation Memory: 8499760640
> [07/07/2023-11:04:53] [I] [TRT] Detected 6 inputs and 1 output network tensors.
> 
all the output information file 
[result.txt](https://github.com/NVIDIA/TensorRT/files/11974209/result.txt)
